cli/cmd/accounts.go

```diff
@@ -376,6 +377,13 @@ var (
 				cobra.CheckErr(err)
 			}
 
+			// Safety check for withdrawals to known secp256k1 accounts.
+			for _, w := range cliConfig.Global().Wallets.All {
+				if w.Address == toAddr.String() && (w.Config["algorithm"] == wallet.AlgorithmSecp256k1Bip44 || w.Config["algorithm"] == wallet.AlgorithmSecp256k1Raw) {
+					cobra.CheckErr(fmt.Errorf("wallet %s uses %s which is not supported on consensus", w.Address, w.Config["algorithm"]))
```

Comment:
Wouldn't it be better if you would add something like `HasConsensusSigner(cfg)` to the wallet factory interface and then just add `w.HasConsensusSigner()` which would load the factory and invoke the method on `w.Config`? The `algorithm` key is an internal detail of the particular wallet implementation and should not be exposed.